File: 04compress/compress.py
# My Solution
def compress(s):
  i = 0
  j = 0
  s += "!"
  result = []
  
  while j < len(s):
    if(s[i] == s[j]):
      j+=1
    else:
      count = j-i
      if count>=2:
        result.append(str(count)+s[i])
        i = j
        j+=1
      else:
        result.append(s[i])
        i = j
        j+=1
  return "".join(result)

print(compress('ccaaatsss'))


# Alvin's solution
def compress(s):
  s += '!'
  result = []
  i = 0
  j = 0
  while j < len(s):
    if s[i] == s[j]:
      j += 1  
    else:
      num = j - i
      if num == 1:
        result.append(s[i])
      else:
        result.append(str(num)) 
        result.append(s[i])
      i = j
    
  return ''.join(result)

File: Structy_2/2_Arrays_Strings/Compress/compress.js
const compress = (s) => {
  // todo
  let result = [];
  let i = 0;
  let j = 0;
  
  while(j <= s.length){    
    if(s[i] === s[j]){
      j += 1;
    }else {
    console.log(result)

      if(j-i > 1){
        result.push(String([j-i]), s[i]) 
      }
      else{
        result.push(s[i])
      }
      i = j
    }
  }
   return result.join("")
};

compress('ccaaatsss'); // -> '2c3at3s'